src/routes/files/handlers.js

```diff
@@ -77,6 +77,13 @@ const hasReportAuthorization = async (user, reportId) => {
   return true;
 };
 
+const reportIsInAnEditableState = async (user, reportId) => {
+  const [report] = await activityReportAndRecipientsById(reportId);
```

Comment:
I don't think you need to call this if you just need to get the status. Maybe just a simple findOne.